$MirOS: ports/mailnews/pine/patches/patch-imap_src_tmail_tmail_c,v 1.2 2005/05/04 23:37:40 tg Exp $
$OpenBSD: patch-tmail_tmail_c,v 1.1 2002/09/18 08:19:06 jakob Exp $
--- imap/src/tmail/tmail.c.orig	Sun Jul 14 22:49:06 2002
+++ imap/src/tmail/tmail.c	Tue Dec  9 13:00:53 2008
@@ -33,7 +33,7 @@ extern int errno;		/* just in case */
 
 /* Globals */
 
-char *version = "2002(14)";	/* tmail release version */
+char *version = "2002(14)L";	/* tmail release version */
 int debug = NIL;		/* debugging (don't fork) */
 int trycreate = NIL;		/* flag saying gotta create before appending */
 int critical = NIL;		/* flag saying in critical code */
@@ -189,7 +189,6 @@ int main (int argc,char *argv[])
       while ((c = getchar ()) != EOF) putc (c,f);
     }
     else {
-      mm_log ("tmail called with LF-only newlines",WARN);
       if ((tmp[0] != 'F') || (tmp[1] != 'r') || (tmp[2] != 'o') ||
 	  (tmp[3] != 'm') || (tmp[4] != ' ')) {
 	*s++ = '\015';		/* overwrite NL with CRLF */
@@ -560,11 +559,10 @@ int fail (char *string,int code)
 
 char *getusername (char *s,char **t)
 {
-  char tmp[MAILTMPLEN];
   if (*t = strchr (s,'+')) {	/* have a mailbox specifier? */
     *(*t)++ = '\0';		/* yes, tie off user name */
 				/* user+ and user+INBOX same as user */
-    if (!**t || !strcmp ("INBOX",ucase (strcpy (tmp,*t)))) *t = NIL;
+    if (!**t || !compare_cstring ((unsigned char *) *t,"INBOX")) *t = NIL;
   }
   return s;			/* return user name */
 }
